components/identity.oauth.uma.permission.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/PermissionApiServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.oauth.uma.endpoint;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.oauth.uma.endpoint.dto.ErrorResponseDTO;
+import org.wso2.carbon.identity.oauth.uma.endpoint.dto.PermissionTicketResponseDTO;
+import org.wso2.carbon.identity.oauth.uma.endpoint.dto.ResourceModelDTO;
+import org.wso2.carbon.identity.oauth.uma.endpoint.dto.ResourceModelInnerDTO;
+import org.wso2.carbon.identity.oauth.uma.endpoint.exception.PermissionEndpointException;
+import org.wso2.carbon.identity.oauth.uma.service.PermissionService;
+import org.wso2.carbon.identity.oauth.uma.service.UMAConstants;
+import org.wso2.carbon.identity.oauth.uma.service.exception.PermissionDAOException;
+import org.wso2.carbon.identity.oauth.uma.service.exception.UMAException;
+import org.wso2.carbon.identity.oauth.uma.service.exception.UMAResourceException;
+import org.wso2.carbon.identity.oauth.uma.service.model.PermissionTicketDO;
+import org.wso2.carbon.identity.oauth.uma.service.model.Resource;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.ws.rs.core.Response;
+
+/**
+ * PermissionApiServiceImpl is used to request permission ticket.
+ */
+public class PermissionApiServiceImpl extends PermissionApiService {
+
+    private static Log log = LogFactory.getLog(PermissionApiServiceImpl.class);
+
+    @Override
+    public Response requestPermission(ResourceModelDTO requestedPermission) {
+
+        PermissionService permissionService = (PermissionService) PrivilegedCarbonContext.getThreadLocalCarbonContext()
+                .getOSGiService(PermissionService.class, null);
+        if (requestedPermission == null) {
+            log.error("Empty request body.");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        PermissionTicketDO permissionTicketDO = null;
+        try {
+            permissionTicketDO = permissionService.issuePermissionTicket(getPermissionTicketRequest(
+                    requestedPermission));
+        } catch (UMAResourceException e) {
+            handleErrorResponse(e, false, log);
+        } catch (PermissionDAOException e) {
+            handleErrorResponse(e, true, log);
+        } catch (Throwable throwable) {
+            handleErrorResponse(throwable, true, log);
+        }
+
+        PermissionTicketResponseDTO permissionTicketResponseDTO = new PermissionTicketResponseDTO();
+        permissionTicketResponseDTO.setTicket(permissionTicketDO.getTicket());
+
+        if (log.isDebugEnabled()) {
+            if (IdentityUtil.isTokenLoggable("permission ticket")) {
```

Comment:
Is this the usual way we name loggable things ? I mean doesn't it need to be like permission_ticket or something camel case ?

Reply:
will change it as PermissionTicket